utils/test_utils.py

```diff
@@ -0,0 +1,52 @@
+import erp
+from django.test import TestCase
+from api.models import Event, FieldReport, Region, Country, DisasterType, ERPGUID
+from unittest import mock
+
+
+class ERPTest(TestCase):
+    def mocked_requests_post(*args, **kwargs):
+        class MockResponse:
+            def __init__(self, json_data, status_code):
+                self.json_data = json_data
+                self.text       = 'FindThisGUID'
+                self.status_code = status_code
+
+            def json(self):
+                return self.json_data
+
+        if not kwargs['json']['Emergency']['FieldReport']['AffectedCountries']:
+            return MockResponse(None, 400)
+        return MockResponse(None, 200)
```

Comment:
Let's define this outside the ERPTest class. This way we can reuse this in `api/test_models.py` as well (keeping a single mock source).

Also, Instead of MockResponse class, we can use `Mock` here. Something like this.
```python3
from unittest.mock import Mock
....

def erp_request_side_effect_mock(url, json):
    def _generate_mock(status_code, json):
        response_mock = Mock()
        response_mock.text = 'FineThisGUID'
        response_mock.status_code = status_code
        response_mock.json.return_value = json
        return response_mock

    if json['Emergency']['FieldReport']['AffectedCountries']:
        return _generate_mock(200, None)
    return _generate_mock(400, None)
```

Reply:
Done.